feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/radio.c

```diff
@@ -460,32 +461,34 @@ bool target_radio_config_set2(const struct schema_Wifi_Radio_Config *rconf,
 {
 	blob_buf_init(&b, 0);
 	blob_buf_init(&del, 0);
-
 	char phy[6];
 	char ifname[8];
-	int freq = 0;
-	char *mode = NULL;
 	struct mode_map *m;
 
 	strncpy(ifname, rconf->if_name, sizeof(ifname));
 	strncpy(phy, target_map_ifname(ifname), sizeof(phy));
 
+	if (full_config && !changed->channel)
+		full_config = false;
+
 	if (changed->channel && rconf->channel) {
 		/* If radio channel config has changed then rebalance channel,
 		 * if not, that means the channel state changed due to a 
 		 * radar detection and shall remain in the backup channel */
 		if (is_cloud_channel_change(rconf) == 0) {
-			if(rrm_radio_rebalance_channel(rconf) == 0) {
+			/* In case of full config push, do not send CSA as hostapd
+			 * is anyway going to reload with reload_config script detecting
+			 * the UCI change. This would help avoid back to back channel switch.
+			 */
+			if (full_config) {
+				LOGD("channel change due to full config push. New channel is %d", rconf->channel);
+				blobmsg_add_u32(&b, "channel", rconf->channel);
+				radio_fixup_set_primary_chan(rconf->if_name, rconf->channel);
+				full_config = false;
+			} else if (rrm_radio_rebalance_channel(rconf) == 0) {
+				LOGD("channel change due to RRM rebalancing. New channel is %d", rconf->channel);
 				blobmsg_add_u32(&b, "channel", rconf->channel);
 				radio_fixup_set_primary_chan(rconf->if_name, rconf->channel);
-
-				/* Update the ht_mode in UCI if we downgraded due to channel change */
-				freq = ieee80211_channel_to_frequency(rconf->channel);
-				mode = get_max_channel_bw_channel(freq,	rconf->ht_mode);
-				if (strncmp(mode, rconf->ht_mode, 4)) {
-					m = mode_map_get_uci(rconf->freq_band, mode, rconf->hw_mode);
-					blobmsg_add_string(&b, "htmode", m->ucihtmode);
-				}
```

Comment:
This will not be needed and with a small change is now taken care in the block starting at line 560. In fact I found an issue with the above code. It does not take care of the "auto" hw_mode and the logic would fail since mode map does not contain "auto" mode. This is again very well taken care in the block starting at line 560.

Reply:
Ok. Makes sense. Thanks for the clarification.